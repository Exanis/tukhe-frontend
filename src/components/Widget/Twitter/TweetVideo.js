import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class TweetVideo extends Component {
    render() {
        const sources = this.props.video.variants.map(video => <source
            key={video.url}
            src={video.url}
            type={video.content_type}
            />
        );

        return <video
            className={this.props.className}
            autoplay="autoplay"
            controls={!this.props.gif}
            loop={this.props.gif}
            muted={false}
            preload="none"
        >{sources}</video>;
    }
}

TweetVideo.propTypes = {
    video: PropTypes.shape({
        variants: PropTypes.arrayOf(PropTypes.shape({
            url: PropTypes.string,
            content_type: PropTypes.string
        }))
    }).isRequired,
    className: PropTypes.string.isRequired,
    gif: PropTypes.bool
};
