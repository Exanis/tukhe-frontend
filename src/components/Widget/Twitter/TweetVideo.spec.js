import React from 'react';
import { shallow } from 'enzyme';
import TweetVideo from './TweetVideo';

const props = {
    video: {
        variants: [{
            url: 'test',
            content_type: 'test content'
        }]
    },
    className: 'test class',
    gif: false
};

it('Should render correctly', () => {
    const wrapper = shallow(<TweetVideo {...props} />);

    expect(wrapper).toMatchSnapshot();
});
