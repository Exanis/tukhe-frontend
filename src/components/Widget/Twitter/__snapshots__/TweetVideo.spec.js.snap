// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TweetVideo
    className="test class"
    gif={false}
    video={
      Object {
        "variants": Array [
          Object {
            "content_type": "test content",
            "url": "test",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "autoplay": "autoplay",
      "children": Array [
        <source
          src="test"
          type="test content"
        />,
      ],
      "className": "test class",
      "controls": true,
      "loop": false,
      "muted": false,
      "preload": "none",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "test",
        "nodeType": "host",
        "props": Object {
          "src": "test",
          "type": "test content",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
    ],
    "type": "video",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "autoplay": "autoplay",
        "children": Array [
          <source
            src="test"
            type="test content"
          />,
        ],
        "className": "test class",
        "controls": true,
        "loop": false,
        "muted": false,
        "preload": "none",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "test",
          "nodeType": "host",
          "props": Object {
            "src": "test",
            "type": "test content",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
      ],
      "type": "video",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
